src/wh_message_custom.c

```diff
@@ -0,0 +1,84 @@
+#include <stddef.h>
+#include <string.h>
+
+#include "wolfhsm/wh_message_custom.h"
+#include "wolfhsm/wh_error.h"
+#include "wolfhsm/wh_comm.h"
+
+
+static void _translateCustomData(uint16_t magic, uint32_t translatedType,
+                                 const whMessageCustom_Data* src,
+                                 whMessageCustom_Data*       dst)
+{
+    if (translatedType < WH_MESSAGE_CUSTOM_TYPE_USER_DEFINED_START) {
+        switch (translatedType) {
+            case WH_MESSAGE_CUSTOM_TYPE_QUERY: {
+                /* right now, no further translations required */
+            } break;
+            case WH_MESSAGE_CUSTOM_TYPE_DMA32: {
+                dst->dma32.client_addr =
+                    wh_Translate32(magic, src->dma32.client_addr);
+                dst->dma32.client_sz =
+                    wh_Translate32(magic, src->dma32.client_sz);
+                dst->dma32.server_addr =
+                    wh_Translate32(magic, src->dma32.server_addr);
+                dst->dma32.server_sz =
+                    wh_Translate32(magic, src->dma32.server_sz);
+            } break;
+            case WH_MESSAGE_CUSTOM_TYPE_DMA64: {
+                dst->dma64.client_addr =
+                    wh_Translate64(magic, src->dma64.client_addr);
+                dst->dma64.client_sz =
+                    wh_Translate64(magic, src->dma64.client_sz);
+                dst->dma64.server_addr =
+                    wh_Translate64(magic, src->dma64.server_addr);
+                dst->dma64.server_sz =
+                    wh_Translate64(magic, src->dma64.server_sz);
+            } break;
+            default: {
+                /* reserved message types - no translation for now */
+            } break;
+        }
+    }
+    else {
+        memcpy(dst->buffer.data, src->buffer.data,
```

Comment:
Consider memmove or a check if src == dst to handle the case there the user passed in the same ptr.  This is probably only called by the server, so low likelihood.